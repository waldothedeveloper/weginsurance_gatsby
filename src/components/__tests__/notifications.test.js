import { render, screen } from "@testing-library/react";

import Notifications from "../notifications";
import React from "react";
import userEvent from "@testing-library/user-event";

const user = userEvent.setup();

//
describe(`Notifications Component`, () => {
  it(`renders the Notifications component`, () => {
    const container = render(
      <Notifications
        authErrors="Error"
        openCloseNotification={true}
        setOpenCloseNotification={() => undefined}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it(`should be unmounted when openCloseNotification is false`, () => {
    const { container, unmount } = render(
      <Notifications
        authErrors="Error"
        openCloseNotification={false}
        setOpenCloseNotification={() => false}
      />
    );

    unmount();

    expect(container).toBeEmptyDOMElement();
  });

  it(`should be unmounted when setOpenCloseNotification is clicked`, async () => {
    const { container, unmount } = render(
      <Notifications
        authErrors="Error"
        openCloseNotification={true}
        setOpenCloseNotification={() => false}
      />
    );

    const link = screen.getByTestId(`close-notification`);

    await user.click(link);

    unmount();

    expect(container).toBeEmptyDOMElement();
  });
});
