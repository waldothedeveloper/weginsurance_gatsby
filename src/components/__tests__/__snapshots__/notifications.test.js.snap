// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Component renders the Notifications component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-live="assertive"
        class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
        data-testid="notifications"
      >
        <div
          class="flex w-full flex-col items-center space-y-4 sm:items-end"
        >
          <div
            class="pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5"
            style="transform: scale(0.9) translateZ(0);"
          >
            <div
              class="p-4"
            >
              <div
                class="flex items-start"
              >
                <div
                  class="flex-shrink-0"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6 text-red-500"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20.618 5.984A11.955 11.955 0 0112 2.944a11.955 11.955 0 01-8.618 3.04A12.02 12.02 0 003 9c0 5.591 3.824 10.29 9 11.622 5.176-1.332 9-6.03 9-11.622 0-1.042-.133-2.052-.382-3.016zM12 9v2m0 4h.01"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
                <div
                  class="ml-3 w-0 flex-1 pt-0.5"
                >
                  <p
                    class="text-sm font-medium text-red-500"
                  >
                    Error!
                  </p>
                  <p
                    class="mt-1 text-sm text-red-500"
                  >
                    Error
                  </p>
                </div>
                <div
                  class="ml-4 flex flex-shrink-0"
                >
                  <button
                    class="inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
                    data-testid="close-notification"
                  >
                    <span
                      class="sr-only"
                    >
                      Close
                    </span>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-live="assertive"
      class="pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
      data-testid="notifications"
    >
      <div
        class="flex w-full flex-col items-center space-y-4 sm:items-end"
      >
        <div
          class="pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5"
          style="transform: scale(0.9) translateZ(0);"
        >
          <div
            class="p-4"
          >
            <div
              class="flex items-start"
            >
              <div
                class="flex-shrink-0"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6 text-red-500"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.618 5.984A11.955 11.955 0 0112 2.944a11.955 11.955 0 01-8.618 3.04A12.02 12.02 0 003 9c0 5.591 3.824 10.29 9 11.622 5.176-1.332 9-6.03 9-11.622 0-1.042-.133-2.052-.382-3.016zM12 9v2m0 4h.01"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <div
                class="ml-3 w-0 flex-1 pt-0.5"
              >
                <p
                  class="text-sm font-medium text-red-500"
                >
                  Error!
                </p>
                <p
                  class="mt-1 text-sm text-red-500"
                >
                  Error
                </p>
              </div>
              <div
                class="ml-4 flex flex-shrink-0"
              >
                <button
                  class="inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
                  data-testid="close-notification"
                >
                  <span
                    class="sr-only"
                  >
                    Close
                  </span>
                  <svg
                    aria-hidden="true"
                    class="h-5 w-5"
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
